Sources/Core/Shared/Operation.swift

```diff
@@ -92,20 +92,11 @@ public class Operation: NSOperation {
         }
         set (newState) {
             willChangeValueForKey("state")
-
-            log.verbose("\(operationName): \(_state) -> \(newState)")
-
-            stateLock.withCriticalScope { () -> Void in
-
-                switch (_state, newState) {
-                case (.Finished, _):
-                    break
```

Comment:
This should actually be considered an invalid transition. As it turns out code never enters this path (which is good) but doesn't help us with the bug.